cmake/scripts/ROOTConfig.cmake.in

```diff
@@ -110,6 +111,16 @@ endforeach()
 #----------------------------------------------------------------------------
 set(ROOT_EXECUTABLE ${ROOT_root_CMD})
 
+#----------------------------------------------------------------------------
+# Deprecation function
+function(__deprecated_var var access)
+  if(access STREQUAL "READ_ACCESS")
+    message(DEPRECATION "The variable '${var}' is deprecated!")
+  endif()
+endfunction()
+
+variable_watch(ROOT_BINARY_DIR __deprecated_var)
```

Comment:
This is only going to complain when `ROOT_BINARY_DIR` is *changed* which shouldn't happen anyway. I'd recommend simply removing it, because it gives the impression that *uses* will be diagnosed when they actually are not.

Reply:
The gist says `set` is called every time it's *used*, so - fine!